functions/issue-scraper.ts

```diff
@@ -109,53 +109,61 @@ interface GraphQlSearchResponse {
 
 export const corsHeaders = {
   "Access-Control-Allow-Origin": "*",
-  "Access-Control-Allow-Methods": "GET",
+  "Access-Control-Allow-Methods": "POST, OPTIONS",
   "Access-Control-Allow-Headers": "Content-Type",
 };
-
-export async function onRequest(ctx: Context): Promise<Response> {
-  const { request, env } = ctx;
+export async function handleIssueScraper(request: Request): Promise<Response> {
   try {
     switch (request.method) {
+      case "OPTIONS": {
+        return new Response(null, { headers: corsHeaders, status: 204 });
+      }
       case "POST": {
         const result = await validatePOST(request);
         if (!result.isValid || !result.gitHubUser) {
-          return new Response("Unauthorized", {
-            headers: corsHeaders,
-            status: 400,
-          });
+          return new Response("Unauthorized", { headers: corsHeaders, status: 401 });
         }
         const githubUserName = result.gitHubUser.login;
         const timestamp = result.timestamp; // Unix timestamp in milliseconds
 
         try {
-          const supabase = new SupabaseClient(env.SUPABASE_URL, env.SUPABASE_KEY);
-          const response = await issueScraper(githubUserName, supabase, env.VOYAGEAI_API_KEY, result.authToken, timestamp);
-          return new Response(response, {
-            headers: corsHeaders,
-            status: 200,
-          });
+          const SUPABASE_URL = Deno.env.get("SUPABASE_URL");
+          const SUPABASE_KEY = Deno.env.get("SUPABASE_KEY");
+          const VOYAGEAI_API_KEY = Deno.env.get("VOYAGEAI_API_KEY");
+
+          // Ensure required environment variables are present in dev/prod
+          const missing: string[] = [];
+          if (!SUPABASE_URL) missing.push("SUPABASE_URL");
+          if (!SUPABASE_KEY) missing.push("SUPABASE_KEY");
+          if (!VOYAGEAI_API_KEY) missing.push("VOYAGEAI_API_KEY");
+          if (missing.length) {
+            return new Response(
+              JSON.stringify({
+                success: false,
+                retryInfo: {
+                  source: "configuration",
+                  status: 503,
+                  retryAfter: 3600,
+                  message: `Service not configured. Missing: ${missing.join(", ")}`,
+                },
+              }),
+              { status: 503, headers: { ...corsHeaders, "Content-Type": "application/json", "Retry-After": "3600" } }
+            );
+          }
+          const supabase = new SupabaseClient(SUPABASE_URL, SUPABASE_KEY);
```

Comment:
This does not exist. It should be
```suggestion
          const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);
```

And it should have typings.